source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1294,20 +1295,29 @@ void RtFileIo::read(const FilePath& documentPath, const FileSearchPath& searchPa
         }
         readFromXmlFile(document, documentPath, searchPaths, &xmlReadOptions);
 
-        string errorMessage;
-        DocumentPtr validationDocument = createDocument();
-        writeUnitDefinitions(validationDocument);
-        CopyOptions cops;
-        cops.skipConflictingElements = true;
-        validationDocument->copyContentFrom(document, &cops);
-        if (validationDocument->validate(&errorMessage))
+        bool accessNodeDefGraphs = !getEnviron("ACCESS_NODEDEF_GRAPHS").empty();
+        if (accessNodeDefGraphs)
```

Comment:
Not sure about the naming of this environment variable? Should it not be named DISABLE_VALIDATION since that is what it's doing? :)

But if there is good reason for supporting disabling of validation in some circumstances I think it would be nicer to add that as an option to RtReadOptions instead? To make this more officially supported. `RtReadOptions.validate` which is true by default.

Reply:
Yes. This was exactly what I was thinking of doing and then the option can be controlled at a higher level :)

Reply:
New interface added now + test